django/utils/datastructures.py

```diff
@@ -131,18 +131,23 @@ def get(self, key, default=None):
             return default
         return val
 
-    def getlist(self, key, default=None):
-        """
-        Returns the list of values for the passed key. If key doesn't exist,
-        then a default value is returned.
-        """
+    def _getlist(self, key, default=None):
+        # Returns mutable list of values for the passed key
+        # Used internally to manipulate list of values
         try:
             return super(MultiValueDict, self).__getitem__(key)
         except KeyError:
             if default is None:
                 return []
             return default
 
+    def getlist(self, key, default=None):
+        """
+        Returns the list of values for the passed key. If key doesn't exist,
+        then a default value is returned.
+        """
+        return list(self._getlist(key, default))
```

Comment:
I don't think there is any requirement for `default` to be a sequence. 

Previously, this code would work:

```python
>>> MISSING = object()
>>> foo = MultiValueDict().getlist('foo', MISSING)
>>> foo is MISSING
True
```

Following the patch:
```python
>>> MISSING = object()
>>> foo = MultiValueDict().getlist('foo', MISSING)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/tmp/venv/src/django/django/utils/datastructures.py", line 149, in getlist
    return list(self._getlist(key, default))
TypeError: 'object' object is not iterable
```

I think we perhaps need to do something like this:
```python
def _getlist(self, key, default=None, force_list=False):
    try:
        values = super(MultiValueDict, self).__getitem__(key)
    except KeyError:
        if default is None:
            return []
        return default
    else:
        if force_list:
            values = list(values)
        return values

def getlist(self, key, default=None):
    """
    Returns the list of values for the passed key. If key doesn't exist,
    then a default value is returned.
    """
    return self._getlist(key, default, True)
````